File: Google's Coding Competitions/Kick Start/2022/Round C/Range_Partition.cpp
// Google Kickstart Round C 2022 - Range Partition
// https://codingcompetitions.withgoogle.com/kickstart/round/00000000008cb4d1/0000000000b20deb

#include "bits/stdc++.h"
using namespace std;

void solve(){
    int N, X, Y;
    cin >> N >> X >> Y;
    long long total_sum = N*(N+1)/2;
    if(total_sum % (X + Y) != 0){
        cout << "IMPOSSIBLE" << endl;
        return;
    }
    cout << "POSSIBLE" << endl;
    long long target = X * total_sum / (X + Y);
    long long curr_sum = 0;

    vector<int> answer;

    long long sum = 0;
    for(int i = N; i > 0; --i){
        if(curr_sum + i <= target){
            curr_sum += i;
            answer.push_back(i);
        }
        if(curr_sum == target){
            break;
        }
    }
    cout << answer.size() << '\n';
    reverse(answer.begin(), answer.end());
    for(int n : answer){
        cout << n << " ";
    }
    cout << '\n';
}


int main() {
    ios::sync_with_stdio(0);
    cin.tie(0);

    int T;
    cin >> T;
    for(int t = 1; t <= T; ++t){
        cout << "Case #" << t << ": ";
        solve();
    }

}

File: Google's Coding Competitions/Kick Start/2022/Round C/Range_Partition.py
# Google Kickstart Round C 2022 - Range Partition
# https://codingcompetitions.withgoogle.com/kickstart/round/00000000008cb4d1/0000000000b20deb

import sys
sys.setrecursionlimit(10**6)
input = sys.stdin.readline

T = int(input())
for T in range(T):
    N, X, Y = map(int, input().split())
    total_sum = N*(N+1)//2
    if total_sum % (X+Y) != 0:
        print(f"Case #{T+1}: IMPOSSIBLE")
        continue
    answer = []
    target = X * total_sum // (X+Y)
    curr_sum = 0
    for i in range(N, 0, -1):
        if curr_sum + i <= target:
            curr_sum += i
            answer.append(i)
        if curr_sum == target:
            break
    answer.reverse()
    sys.stdout.write(f"Case #{T+1}: POSSIBLE\n{len(answer)}\n")
    print(*answer)
    